TypeScript error when require eslint-plugin-jsdoc in a CommonJS file

## Expected behavior

Able to use TypeScript for type checking in a CommonJS file which imports `eslint-plugin-jdocs` (using `require`).

## Actual behavior

TypeScript fails on the require line for `eslint-plugin-jdocs` in a CommonJS file. 

```console
$ tsc -p tsconfig.js.json

eslint.config.js:6:23 - error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("eslint-plugin-jsdoc")' call instead.
  To convert this file to an ECMAScript module, change its file extension to '.mjs' or create a local package.json file with `{ "type": "module" }`.

6 const jsdoc = require('eslint-plugin-jsdoc');
```

My understanding of the problem is that TypeScript best practice is a definition file corresponds to a single JavaScript file. I am guessing when TypeScript finds the `types` file for  eslint-plugin-jdocs  of `./dist/index.d.ts`, it assumes the corresponding JavaScript file will be the esm flavour since eslint-plugin-jdocs has `"type": "module"` and the `.d.ts` is therefore implicitly esm. 

There is a project which champions technically correct TypeScript setups and picks up a potential problem. This might be more convincing than my scenario. 😄 

- https://arethetypeswrong.github.io/?p=eslint-plugin-jsdoc%4048.2.3
- https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/main/docs/problems/FalseESM.md

Quoting from [FalseESM](https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/main/docs/problems/FalseESM.md):

> A golden rule of declaration files is that if they represent a module—that is, if they use `import` or `export` at the top level—they must represent _exactly_ one JavaScript file. They _especially_ cannot represent JavaScript files of two different module formats. 

I was able to fix my error without actually creating a separate definition file by modifying eslint-plugin-jdocs `package.json` file like this, but I suspect it is not robust so more for interest than a suggestion!

```
  "exports": {
    "import": {
      "types": "./dist/index.d.ts",
      "default": "./src/index.js"
    },
    "require": {
      "types": "./dist/index.d.ts",
      "default": "./dist/index.cjs"
    }
  },
```

I assume the robust solution would be to have an explicit definition file for cjs (say): `index.d.cts`. 

TypeScript is normally fairly forgiving about this problem to avoid breaking consumers (https://github.com/microsoft/TypeScript/issues/50762#issuecomment-1528318260) and I seem to have stumbled on a case where it notices.

Happy to work on a simple reproduction if you would like to see TypeScript failure yourself rather than just the `arethetypeswrong` warning, and willing to work on a PR.

## Environment

- Node version: v20.11.1
- ESLint version v8.56.0
- `eslint-plugin-jsdoc` version: 48.2.3
- TypeScript version: 5.2.2

--- a/package.json
+++ b/package.json
@@ -90,19 +90,29 @@
         "default": "./src/index.js"
       },
       "require": {
-        "types": "./dist/index-cjs.d.ts",
-        "default": "./dist/index-cjs.cjs"
+        "types": "./dist/cjs/index.cjs.d.cts",
+        "default": "./dist/cjs/index.cjs.cts"
       }
     },
     "./getJsdocProcessorPlugin.js": {
-      "types": "./dist/getJsdocProcessorPlugin.d.ts",
-      "import": "./src/getJsdocProcessorPlugin.js",
-      "require": "./dist/getJsdocProcessorPlugin.cjs"
+      "import": {
+        "types": "./dist/getJsdocProcessorPlugin.d.ts",
+        "default": "./src/getJsdocProcessorPlugin.js"
+      },
+      "require": {
+        "types": "./dist/cjs/getJsdocProcessorPlugin.d.cts",
+        "default": "./dist/cjs/getJsdocProcessorPlugin.cjs"
+      }
     },
     "./iterateJsdoc.js": {
-      "types": "./dist/iterateJsdoc.d.ts",
-      "import": "./src/iterateJsdoc.js",
-      "require": "./dist/iterateJsdoc.cjs"
+      "import": {
+        "types": "./dist/iterateJsdoc.d.ts",
+        "default": "./src/iterateJsdoc.js"
+      },
+      "require": {
+        "types": "./dist/cjs/iterateJsdoc.d.cts",
+        "default": "./dist/cjs/iterateJsdoc.cjs"
+      }
     }
   },
   "name": "eslint-plugin-jsdoc",
@@ -145,7 +155,8 @@
     "ruleTypes": "node ./src/bin/generateRuleTypes.js",
     "tsc": "tsc",
     "tsc-build": "tsc -p tsconfig-prod.json",
-    "build": "node ./src/bin/buildEntryFileForTS.js && rimraf ./dist && NODE_ENV=production babel ./src --out-file-extension .cjs --out-dir ./dist --copy-files --source-maps --ignore ./src/bin/*.js --no-copy-ignored && replace 'require\\(\"\\.(.*?)\\.[^.]*?\"\\)' 'require(\".$1.cjs\")' 'dist' -r --include=\"*.cjs\" && pnpm tsc-build",
+    "tsc-cjs": "tsc -p tsconfig-cjs.json",
+    "build": "node ./src/bin/buildEntryFileForTS.js && rimraf ./dist && NODE_ENV=production babel ./src --out-file-extension .cjs --out-dir ./dist --copy-files --source-maps --ignore ./src/bin/*.js --no-copy-ignored && replace 'require\\(\"\\.(.*?)\\.[^.]*?\"\\)' 'require(\".$1.cjs\")' 'dist' -r --include=\"*.cjs\" && pnpm tsc-build && pnpm tsc-cjs",
     "attw": "attw --pack .",
     "check-docs": "node ./src/bin/generateDocs.js --check",
     "create-docs": "pnpm run create-options && node ./src/bin/generateDocs.js",

--- a/src/getJsdocProcessorPlugin.cts
+++ b/src/getJsdocProcessorPlugin.cts
@@ -1,0 +1,3 @@
+import {getJsdocProcessorPlugin} from './getJsdocProcessorPlugin.js';
+
+exports.getJsdocProcessorPlugin = getJsdocProcessorPlugin;

--- a/src/index.cjs.cts
+++ b/src/index.cjs.cts
@@ -1,0 +1,3 @@
+import items from './index-cjs.js';
+
+export = items;

--- a/src/iterateJsdoc.cts
+++ b/src/iterateJsdoc.cts
@@ -1,0 +1,6 @@
+import iterateJsdoc, {getSettings, parseComment} from './iterateJsdoc.js';
+
+export = iterateJsdoc;
+
+exports.getSettings = getSettings;
+exports.parseComment = parseComment;

--- a/tsconfig-cjs.json
+++ b/tsconfig-cjs.json
@@ -1,0 +1,16 @@
+{
+  "compilerOptions": {
+    "target": "es2023", // Or your desired target
+    "module": "commonjs", // Crucial for CommonJS output
+    "allowJs": true,
+    "declaration": true, // Generates declaration files (.d.ts)
+    "outDir": "./dist/cjs", // Output directory for CommonJS files
+    "esModuleInterop": true, // Improves interoperability between ESM and CJS
+    "forceConsistentCasingInFileNames": true,
+    "strict": true,
+    "skipLibCheck": true
+  },
+  "include": [
+    "src/*.cts"
+  ] // Your source files
+}

